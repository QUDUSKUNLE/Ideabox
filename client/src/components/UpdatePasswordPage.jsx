import React from 'react';
import UpdatePasswordForm from './home/UpdatePasswordForm';

/**
 * @description - renders LogInForm Component
 * @class UpdatePasswordPage
 * @extends {React.Component}
 */
export default class UpdatePasswordPage extends React.Component {
  /**
   * Create a constructor
   * @constructor
   * @param {object} props
   * @memberof UpdatePasswordPage
   */
  constructor(props) {
    super(props);
    this.state = {};
  }

  /**
  * @description - render method, React lifecycle method
  * @returns {object} UpdatePasswordPage component
  */
  render() {
    return (
      <div className="container">
        <div className="logo">
          <h4>IdeaBox</h4>
        </div>
        <div id="update_password" className="align-center">
          <UpdatePasswordForm
            hash={this.props}
          />
        </div>
      </div>
    );
  }
}

// UpdatePasswordPage.propTypes = {
//   history: PropTypes.object.isRequired
// };
// export default HomePage;
