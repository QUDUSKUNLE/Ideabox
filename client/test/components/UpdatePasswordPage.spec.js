import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import UpdatePasswordPage from '../../src/components/UpdatePasswordPage';

Enzyme.configure({ adapter: new Adapter() });

describe('<UpdatePasswordPage />', () => {
  it('component should be defined', () => {
    expect(UpdatePasswordPage).toBeDefined();
  });

  it('should render correctly', () => {
    const component = shallow(<UpdatePasswordPage />);
    expect(component).toMatchSnapshot();
    expect(component.find('div')).toHaveLength(3);
  });
});
