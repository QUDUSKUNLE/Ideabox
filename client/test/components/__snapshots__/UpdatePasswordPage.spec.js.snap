// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdatePasswordPage /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UpdatePasswordPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo"
        >
          <h4>
            IdeaBox
          </h4>
        </div>,
        <div
          className="align-center"
          id="update_password"
        >
          <UpdatePasswordForm
            hash={Object {}}
          />
        </div>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4>
            IdeaBox
          </h4>,
          "className": "logo",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "IdeaBox",
          },
          "ref": null,
          "rendered": "IdeaBox",
          "type": "h4",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <UpdatePasswordForm
            hash={Object {}}
          />,
          "className": "align-center",
          "id": "update_password",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "hash": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo"
          >
            <h4>
              IdeaBox
            </h4>
          </div>,
          <div
            className="align-center"
            id="update_password"
          >
            <UpdatePasswordForm
              hash={Object {}}
            />
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4>
              IdeaBox
            </h4>,
            "className": "logo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "IdeaBox",
            },
            "ref": null,
            "rendered": "IdeaBox",
            "type": "h4",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <UpdatePasswordForm
              hash={Object {}}
            />,
            "className": "align-center",
            "id": "update_password",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "hash": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
